rfc/2019-12-07-fuzz-and-auto-debug.md

```diff
@@ -0,0 +1,161 @@
+# Proposal: Fuzz and Auto Debug Tool
+
+- Author(s):     [DQinYuan](https://github.com/DQinYuan)
+- Last updated:  2019-12-07
+- Discussion at:  [issue#124](https://github.com/pingcap/community/issues/124)
+
+## Abstract
+
+There is a lot of uncertainty in Chaos and Fuzz test of complicated software like TiDB. Developers often costs a few of weeks to localize the root cause. This proposal may help on this with fuzz mutation and auto debug.
+
+
+
+## Background
+
+There are many troublesome bugs,  you don't know what's going wrong, even worse, you can not reproduce it .  
+
+ 
+
+[Statistical debug](https://stackoverflow.com/questions/505907/what-is-statistical-debugging) is a common auto debug idea, but it need a lot of test cases to accurately localize bug position.
+
+
+
+Fuzz test is a good idea to apply unlimited case to application, it will provide a bulk of cases to Statistical Debug.  Mutation-Based Fuzzing can introduce small changes  to existing inputs that may still keep the input valid. Some paper guide mutations with some  algorithms to converge input in a root bug. 
```

Comment:
```suggestion
Fuzz test is a good idea to apply unlimited cases to application, it will provide a bulk of cases to Statistical Debug. Mutation-Based Fuzzing can introduce small changes to existing inputs that may still keep the input valid. Some paper guide mutations with some algorithms to converge input in a root bug. 
```